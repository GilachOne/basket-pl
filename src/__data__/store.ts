import { configureStore } from '@reduxjs/toolkit';
import { setupListeners } from '@reduxjs/toolkit/query';

import { api } from './api/main';
import { authApi } from './api/auth';
import { reducer as errorTextReducer } from './slices/errorTextSlice';

export const store = configureStore({
    reducer: {
        [api.reducerPath]: api.reducer,
        [authApi.reducerPath]: authApi.reducer,

        errorText: errorTextReducer,
    },

    middleware: getDefaultMiddleware =>
        getDefaultMiddleware()
            .concat(api.middleware)
            .concat(authApi.middleware),
});

setupListeners(store.dispatch);

// Infer the `RootState` and `AppDispatch` types from the store itself
export type RootState = ReturnType<typeof store.getState>;
