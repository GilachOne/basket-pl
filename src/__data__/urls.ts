import { getNavigationsValue } from '@ijl/cli';

const baseURL = getNavigationsValue('basket.main');

export const URLs = {
    landing: {
        url: `${baseURL}`,
        isEnabled: getNavigationsValue('basket.main'),
    },
    dashboard: {
        url: `${baseURL}${getNavigationsValue('basket.dashboard')}`,
        isEnabled: getNavigationsValue('basket.dashboard'),
    },
    signIn: {
        url: `${baseURL}${getNavigationsValue('basket.sign.in')}`,
        getUrlWidthEmail: email => {
            if (email) {
                return `${baseURL}${getNavigationsValue(
                    'basket.sign.in'
                )}?email=${email}`;
            }
            return `${baseURL}${getNavigationsValue('basket.sign.in')}`;
        },
        isEnabled: getNavigationsValue('basket.sign.in'),
    },
    signUp: {
        url: `${baseURL}${getNavigationsValue('basket.sign.up')}`,
        getUrlWidthEmail: email => {
            if (email) {
                return `${baseURL}${getNavigationsValue(
                    'basket.sign.up'
                )}?email=${email}`;
            }
            return `${baseURL}${getNavigationsValue('basket.sign.up')}`;
        },
        isEnabled: getNavigationsValue('basket.sign.up'),
    },
    list: {
        url: `${baseURL}${getNavigationsValue('basket.list')}`,
        getUrl: listId =>
            `${baseURL}${getNavigationsValue('basket.list')}`.replace(
                ':id',
                listId
            ),
        isEnabled: getNavigationsValue('basket.list'),
    },
    add: {
        url: `${baseURL}${getNavigationsValue('basket.add')}`,
        getUrl: listId =>
            `${baseURL}${getNavigationsValue('basket.add')}`.replace(
                ':id',
                listId
            ),
        isEnabled: getNavigationsValue('basket.add'),
    },
    resetPassword: {
        url: `${baseURL}${getNavigationsValue('basket.reset.password')}`,
        isEnabled: getNavigationsValue('basket.reset.password'),
    },
};
