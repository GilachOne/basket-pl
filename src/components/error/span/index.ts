import styled from '@emotion/styled';

export const ErrorSpan = styled.span`
    color: ${props => props.theme.colors.error.main};
    font-family: 'Roboto';
    font-style: normal;
    font-weight: 500;
    font-size: 14px;
	margin-bottom:24px;
    display:inline-block;
`;
