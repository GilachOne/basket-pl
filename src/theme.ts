export const theme = {
    colors: {
        brand: {
            light2: '#08DA36',
            light: '#74D14C',
            main: '#74D14C',
            dark: '#54AB26',
        },
        background: {
            light: '#FFFFFF',
			main: '#008060',
        },
        accent: {
            light6: '#CCCCCC',
            light5: '#F6EAFF',
            light4: '#d2d8e0',
            light3: '#FAF9F8',
            light2: '#D2D5D9',
            light: '#F6EAFF',
            main: '#9828F0',
            dark: '#2C107A',
        },
        text: {
            light3: '#A68FE5',
            light2: '#42474C',
            light: '#6B7177',
            main: '#FFFFFF',
            dark: '#000000',
        },
        bg: {
            main: '#000000',
        },
        error: {
            main: '#FF0000',
        },
    },
};
