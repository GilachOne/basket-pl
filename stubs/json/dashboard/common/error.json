{
    "success": false,
    "error": "Не получилось..."
}