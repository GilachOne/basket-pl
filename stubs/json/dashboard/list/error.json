{
    "success": false,
    "error": "Список не загрузился"
}