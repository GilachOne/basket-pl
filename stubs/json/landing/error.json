{
	"success": false,
	"error": "Ошибка получения данных для Landing"
}