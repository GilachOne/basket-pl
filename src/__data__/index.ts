import oopss from './animations/oopss.json';
import error from './animations/error.json';

export const animations = {
    oopss,
    error,
};
