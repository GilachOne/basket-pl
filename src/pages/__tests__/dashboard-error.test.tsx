import React from 'react';
import { describe, it, expect, beforeAll, afterAll } from '@jest/globals';
import { render, screen, waitFor } from '@testing-library/react';
import { ThemeProvider } from '@emotion/react';
import { Provider } from 'react-redux';
import { setupServer } from 'msw/node';
import { rest } from 'msw';
import 'whatwg-fetch';
import { BrowserRouter } from 'react-router-dom';

import { theme } from '../../theme';
import { store } from '../../__data__/store';
import DashboardPage from '../dashboard/dashboard';
import error from '../../../stubs/json/dashboard/list/error.json';

const server = setupServer(
    rest.get('/api/dashboard/list', (req, res, ctx) => {
        return res(ctx.status(400), ctx.json(error));
    })
);

describe('Dashboard Page', () => {
    beforeAll(() => {
        server.listen();
    });
    afterAll(() => {
        server.close();
    });

    it('Dashboard Error Test', async () => {
        const { container } = render(
            <Provider store={store}>
                <ThemeProvider theme={theme}>
                    <BrowserRouter>
                        <DashboardPage />
                    </BrowserRouter>
                </ThemeProvider>
            </Provider>
        );

        await waitFor(() => screen.getByText(/Создай/i));
        expect(container).toMatchSnapshot();
    });
});
