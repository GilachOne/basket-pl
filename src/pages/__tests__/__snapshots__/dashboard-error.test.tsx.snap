// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard Page Dashboard Error Test 1`] = `
<div>
  <div
    class="css-1wxw69z"
  >
    <header
      class="css-22fkfx"
    >
      <p
        class="css-1adnaza"
      >
        вМагазин
      </p>
      <div
        class="css-1v5wvrd"
      >
        <button
          class="css-z414q7"
          data-testid="Выход"
        >
          <img
            src="log-out.png"
            title="Выход"
          />
        </button>
      </div>
    </header>
    <div
      class="css-0"
    >
      <div
        class="css-1sie743"
      >
        Список не загрузился
      </div>
      <main
        class="css-1jsn5n8"
      >
        <img
          alt="logo"
          class="css-wnwrug"
          src="logo.png"
        />
        <p
          class="css-f4jk5p"
        >
          Создайте свой первый список.
        </p>
        <p
          class="css-f4jk5p"
        >
          Сделайте покупки быстрее и удобнее
        </p>
        <button
          class="css-ftp01x"
          data-testid="addButton"
        >
          +
        </button>
      </main>
    </div>
  </div>
</div>
`;
